### ISSUE TITLE
Where date = new Date() returns data from the day before

### ISSUE THREAD

**Issue by tombohub** [kysely-org/kysely#527]
I have this query:

```javascript
const rows = await db
    .selectFrom("overnight.daily_overnight_performance")
    .select(({ fn }) => fn.countAll().as("count"))
    .where("change_pct", ">", "0")
    .where("date", "=", new Date("2023-06-02"))
    .execute();
```

and result is `[ { count: '5203' } ]`

sql query result using `compile()` is:

```sql
sql: 'select count(*) as "count" from "overnight"."daily_overnight_performance" 
where "change_pct" > $1 and "date" = $2',
  parameters: [ '0', 2023-06-02T00:00:00.000Z ]
```

Running the same query in database with this query:

```sql
select count(*) from overnight.daily_overnight_performance dop 
where change_pct > '0'
and "date" = '2023-06-02T00:00:00.000Z';
```

returns 

|count|
|-----|
|7167|

Not the same result right?

So I've tried day earlier in database with query:

```sql
select count(*) from overnight.daily_overnight_performance dop 
where change_pct > '0'
and "date" = '2023-06-01T00:00:00.000Z';

```

results in 

|count|
|-----|
|5203|

which is the now same as kysely result.

Why is that?

And why we cannot use string in `where('date', '=', '2023-06-02')` ?

**Comment by koskimas** [comment]
This is probably caused by timezones. The date you create using `new` gets put to the local timezone, which switches its date.

**Comment by tombohub** [comment]
I don't understand your answer and closing the issue. So compile method prints out the wrong SQL? If I run the same SQL kysely prints and the same in database which is in my timezone and get different results then something is wrong. Is there a solution? 

Regarding the date string it's the kysely-codegen generated interface. It - doesn't generate Date string, but Timestamp

**Comment by koskimas** [comment]
It's not the same input. You are trusting some `console.log` or debugger output, which shows you the date as an UTC ISO date. In your runtime environment it actually has a timezone.

> I don't understand your answer and closing the issue.

Well too bad 😄 I'm helping you for free on my spare time. And this is the end of my help. Deal with it. If you don't like it, don't use Kysely.
